app/code/Magento/SalesRule/Model/ResourceModel/Rule/DeleteCustomerUsage.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\SalesRule\Model\ResourceModel\Rule;
+
+
+class DeleteCustomerUsage
+{
+    /** @var \Magento\SalesRule\Model\ResourceModel\Rule\Customer */
+    protected $_customerRuleDetails;
+
+    /**
+     * DeleteCustomerUsage constructor.
+     * @param Customer $customerRuleDetails
+     */
+    public function __construct(
+        Customer $customerRuleDetails
+    )
+    {
+        $this->_customerRuleDetails = $customerRuleDetails;
+    }
+
+    /**
+     * Delete the time Usage from salesrule_customer table when times_used is 0
+     * @param int $ruleId
+     * @param int $customerId
+     * @param int $updatedTimeUsed
+     */
+    public function deleteCustomerTimeUsage($ruleId, $customerId, $updatedTimeUsed)
```

Comment:
Since the name of the class `DeleteCustomerUsage` is service-oriented and conveys the idea of the purpose of the class, what about renaming the `deleteCustomeTimeUsage()` method into `execute()`?

Reply:
@aleron75 ,
I have changed the method from deleteCustomeTimeUsage() to execute(). Kindly check and let me know.